src/Persistence/SQL.php

```diff
@@ -453,7 +453,7 @@ public function _typecastSaveField(Field $f, $value)
             $dt_class = isset($f->dateTimeClass) ? $f->dateTimeClass : 'DateTime';
             $tz_class = isset($f->dateTimeZoneClass) ? $f->dateTimeZoneClass : 'DateTimeZone';
 
-            if ($v instanceof $dt_class) {
+            if ($v instanceof $dt_class || $v instanceof \DateTimeInterface) {
```

Comment:
Few lines below we call `setTimezone` method, but DateTimeInterface doesn't support that.
https://www.php.net/manual/en/class.datetimeinterface.php
Probably need some additional condition in line 461 ( maybe `&& $v instanceof $dt_class`)

Reply:
That is correct, there was also a mistake when the class was immutable (introduced in the past). Wdyt about the new fix which fixes both?